code/script.js

```diff
@@ -251,48 +255,73 @@ const setSecret = () => {
-  // What else should happen when we start the game?
+  board.style.display = 'flex'
+  winOrLose.style.display = 'none'
+  questions.selectedIndex = defaultStatus
+  generateBoard()
+  setSecret()
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+// Setting the currentQuestion object when you select something in the dropdown
+// ! It doesn't work with shorthands for some reason. I know I could use it here though !
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
+  const value = questions.options[questions.selectedIndex].value
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      value: value,
       category: category,
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: value,
+      category: category,
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      attribute: value,
+      value: true,
+      category: category,
+    }
+  } else {
+    currentQuestion = {
+      attribute: value,
+      value: true,
       category: category,
     }
-  } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
   }
 }
 
 // This function should be invoked when you click on 'Find Out'.
-const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-}
+const checkQuestion = (selectQuestion) => {
+  let keep
 
-// It'll filter the characters array and redraw the game board.
+  if (currentQuestion.category === 'hair color') {
+      keep = currentQuestion.value === secret.hairColor 
+  } else if (currentQuestion.category === 'eye color') {
+      keep = currentQuestion.value === secret.eyeColor
+  } else if (currentQuestion.category === 'accessories') {
+    if (currentQuestion.attribute === 'glasses') {
+      keep = secret.glasses === true
+    } else {
+      keep = secret.hat === true
+    }
+  } else {
+      keep = secret.smoker === true
+  }
+
+  filterCharacters(keep)
+} 
+
+// Filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
+  let group = currentQuestion.category
+  const attribute = currentQuestion.attribute
+
   if (group === 'accessories') {
```

Comment:
These nested ifs are a bit hard to read. I know that the project was structured this way to begin with (and in the beginning, I also had these nested ifs in my code), and as your project is working perfectly, it's good the way it is. 
In case you want to rewrite your code, I would like to share that someone on Slack posted a really nice solution on how to make this less messy. If you give the objects created inside selectQuestion another property, text, and store the innerText of the select menu options as the value of text (so that for eyecolor, it can say "brown eyes"), you can use the text property to fill in the alert text. And that way, you only need one alert for yes and one for no. If you want to, you can have a look at my solution. Not sure if I explained it well - let me know if you have questions about that.

Reply:
`switch` would be a better way to write this cleaner 🤓